apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/SchemaDownloader.kt

```diff
@@ -39,30 +44,56 @@ object SchemaDownloader {
             addHeader(it.key, it.value)
           }
         }
-        .url(endpoint)
+        .url(url)
         .build()
 
-    val response = OkHttpClient.Builder()
-        .connectTimeout(connectTimeoutSeconds, TimeUnit.SECONDS)
-        .readTimeout(readTimeoutSeconds, TimeUnit.SECONDS)
-        .build()
-        .newCall(request).execute()
+    val response = newOkHttpClient()
+        .newCall(request)
+        .execute()
 
     if (!response.isSuccessful) {
       throw Exception("cannot get schema: ${response.code}:\n${response.body?.string()}")
     }
 
-    schema.parentFile?.mkdirs()
+    return response
+  }
+
+  fun downloadIntrospection(
+      endpoint: String,
+      headers: Map<String, String>
+  ): String {
+
+    val response = executeQuery(introspectionQuery, endpoint, headers)
+
+    return response.body.use { responseBody ->
+      responseBody!!.string()
+    }
+  }
 
-    response.body.use { responseBody ->
-      if (schema.extension.toLowerCase() == "json") {
-        schema.writeText(responseBody!!.string())
-      } else {
-        IntrospectionSchema(responseBody!!.byteStream()).toSDL(schema)
+  fun downloadRegistry(graph: String, key: String, variant: String): String? {
+    val registryQuery = """
+    query {
+      service(id: "$graph") {
+        schema(tag: "$variant") {
+          document
+        }
       }
     }
+  """.trimIndent()
+
+    val response = executeQuery(registryQuery, "https://engine-graphql.apollographql.com/api/graphql", mapOf("x-api-key" to key))
```

Comment:
can we add in an `apollographql-client-name` header? This will help us make sure that we're tracking this operation back to this particular client so if there are any changes that might affect it down the road, we can reach out :) https://www.apollographql.com/docs/studio/client-awareness/

Reply:
Sounds good 👍 . I also added `apollographql-client-version` there just in case.

Reply:
Aslo, I set the client name to `apollo-gradle-plugin`, which is the name of the maven artifact. Let me know how that sounds.